### ISSUE TITLE
Postgres : SingleSelect/Multiselect isn't working in 0.84.9 

### ISSUE THREAD

**Issue by o1lab** [nocodb/nocodb#1132]
Was fine in earlier version.

**Comment by wingkwong** [comment]
Extracted from Discord.

> Seems like the new version broke at least the SingleSelect option. The columns created pre-update still highlight the options with colored background etc. But for new columns and old columns that I edit, it seems like I can no longer set SingleSelect options. I add the options, click save, refresh the page and the options are gone. Also downgrading to version 0.84.8, makes it work properly again.

Steps to reproduce:
1. Create a new table
2. Add a new column of type SingleSelet or MultiSelect
3. Add few options
4. Click save
5. Refresh the page.
6. All the options are gone.

```
Node: **v12.22.6**
Arch: **x64**
Platform: **linux**
Docker: **true**
Database: **pg**
ProjectOnRootDB: **true**
RootDB: **pg**
PackageVersion: **0.84.9**
```

**Comment by wingkwong** [comment]
Findings:
- ``dtxp`` got overwritten in latest implementation if user create a SingleSelect / MultiSelect with datatype anything other than set / enum

Things to change to persist ``dtxp``:
- RestApiBuilder.ts - onTableCreate()
- GqlApiBuilder.ts - onTableCreate()
- BaseApiBuilder.ts - onTableUpdate()

**Comment by wingkwong** [comment]
Fixed in ``0.84.10``.
